sdk/cosmos/src/clients/database_client.rs

```diff
@@ -49,11 +51,28 @@ impl DatabaseClient {
     }
 
     /// Create a collection
-    pub fn create_collection<'a, P: Into<PartitionKey>>(
-        &'a self,
-        partition_key: P,
-    ) -> requests::CreateCollectionBuilder<'a> {
-        requests::CreateCollectionBuilder::new(self, partition_key.into())
+    pub async fn create_collection<S: AsRef<str>>(
+        &self,
+        ctx: Context,
+        collection_name: S,
+        options: CreateCollectionOptions,
```

Comment:
Same as above, I think `create_collection::Options` would be less verbose and flexible.